samples/augment-reality-to-show-hidden-infrastructure/src/main/java/com/esri/arcgismaps/sample/augmentrealitytoshowhiddeninfrastructure/screens/AugmentedRealityScreen.kt

```diff
@@ -0,0 +1,369 @@
+            }
+            DropdownMenu(
+                expanded = actionsExpanded, onDismissRequest = { actionsExpanded = false }) {
+                DropdownMenuItem(text = { Text("World tracking") }, onClick = {
+                    trackingMode = WorldScaleTrackingMode.World()
+                    actionsExpanded = false
+                })
+                DropdownMenuItem(text = { Text("Geospatial tracking") }, onClick = {
+                    trackingMode = WorldScaleTrackingMode.Geospatial()
+                    actionsExpanded = false
+                })
+            }
+        })
+    }, content = {
+        if (showPrivacyInfo) {
+            PrivacyInfoDialog(
+                hasCurrentlyAccepted = acceptedPrivacyInfo,
+                onUserResponse = { accepted ->
+                    acceptedPrivacyInfo = accepted
+                    sharedPreferences.edit { putBoolean(KEY_PREF_ACCEPTED_PRIVACY_INFO, accepted) }
+                    showPrivacyInfo = false
+                }
+            )
+        }
+        if (!acceptedPrivacyInfo) {
+            Column(
+                modifier = Modifier.fillMaxSize(),
+                verticalArrangement = Arrangement.Center,
+                horizontalAlignment = Alignment.CenterHorizontally
+            ) {
+                Text(text = "Privacy Info not accepted")
+                Button(onClick = { showPrivacyInfo = true }) {
+                    Text(text = "Show Privacy Info")
+                }
+            }
+        } else {
+            Box(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(it),
+            ) {
+                WorldScaleSceneView(
+                    modifier = Modifier.fillMaxSize(),
+                    arcGISScene = augmentedRealityViewModel.arcGISScene,
+                    graphicsOverlays = listOf(
+                        augmentedRealityViewModel.pipeGraphicsOverlay,
+                        augmentedRealityViewModel.pipeShadowGraphicsOverlay,
+                        augmentedRealityViewModel.leaderGraphicsOverlay
+                    ),
+                    worldScaleTrackingMode = trackingMode,
+                    onInitializationStatusChanged = { status ->
+                        initializationStatus = status
+                    }) {
+                    Box(modifier = Modifier.fillMaxSize()) {
+                        if (trackingMode is WorldScaleTrackingMode.World) {
+                            if (displayCalibrationView) {
+                                CalibrationView(
+                                    onDismiss = { displayCalibrationView = false },
+                                    modifier = Modifier.align(Alignment.BottomCenter),
+                                )
+                            }
+                        }
+                    }
+                }
+
+                when (val status = initializationStatus) {
```

Comment:
There a lot going on in this one Composable `AugmentedRealityScreen`, maybe consider breaking it down to individual Composables and building them in the Scaffold:  PrivacyInfoDialog which handles not accepted info, WorldScaleSceneView, WorldScaleSceneViewStatusHandler, FloatingActionButtonOptions.